### PR TITLE
refactor attacher to use new api objects

### PR DESCRIPTION
**Pull request by sbezverk**
Refactor attacher to use latest CSI spec changes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM alpine
+FROM alpine
```
Comment (jsafrane) [inline Dockerfile]: Remove extra space

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-FROM alpine
+FROM alpine 
 LABEL maintainers="Kubernetes Authors"
 LABEL description="CSI External Attacher"
 
-COPY csi-attacher csi-attacher
+COPY ./deploy/docker/csi-attacher csi-attacher
```
Comment (jsafrane) [inline Dockerfile]: in Makefile you put the binary into `./bin/csi-attacher`. Who copies it to `/deploy/docker/csi-attacher`?

File: vendor/github.com/container-storage-interface/spec/lib/go/csi/v0/csi.pb.go [hunk 3]
```diff
@@ -0,0 +1,2671 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (jsafrane) [inline vendor/github.com/container-storage-interface/spec/lib/go/csi/v0/csi.pb.go]: Should be old `container-storage-interface/spec/lib/go/csi/csi.pb.go` removed?

File: Dockerfile (hunk 2 above)
Comment (jsafrane) [reply Dockerfile]: In other words, Dockerfile should probably use directly `bin/csi-attacher`, without copying it anywhere else.

**Comment by sbezverk** [comment]
@jsafrane Thank you for your review, I will address all your comments. With regards to failing tests, there is not much I can do at this point as csi-test and GoCSI are not yet refactored to support new CSI spec version. I expect once it is done, testing should stop failing. I will keep `WIP` until that time.

File: Dockerfile (hunk 2 above)
Comment (sbezverk) [reply Dockerfile]: done

File: vendor/github.com/container-storage-interface/spec/lib/go/csi/v0/csi.pb.go (hunk 3 above)
Comment (sbezverk) [reply vendor/github.com/container-storage-interface/spec/lib/go/csi/v0/csi.pb.go]: removed

**Comment by sbezverk** [comment]
@jsafrane tests are green now PTAL.

**Comment by jsafrane** [comment]
Thanks!

**Pull request merged**